src/util/setSelectionsAndFocusEditor.ts

```diff
@@ -46,48 +55,47 @@ export async function focusEditor(editor: TextEditor) {
   } else {
     // If the view column is null we see if it's a notebook and try to see if we
     // can just move around in the notebook to focus the correct editor
-    const activeTextEditor = window.activeTextEditor;
 
-    if (activeTextEditor == null) {
-      return;
+    if (lt(version, "1.68.0")) {
+      return await focusNotebookCellLegacy(editor);
     }
 
-    const editorNotebook = getNotebookFromCellDocument(editor.document);
-    const activeEditorNotebook = getNotebookFromCellDocument(
-      activeTextEditor.document
-    );
+    await focusNotebookCell(editor);
+  }
+}
 
-    if (
-      editorNotebook == null ||
-      activeEditorNotebook == null ||
-      editorNotebook !== activeEditorNotebook
-    ) {
-      return;
-    }
+async function focusNotebookCell(editor: TextEditor) {
+  const desiredNotebookEditor = getNotebookFromCellDocument(editor.document);
+  if (desiredNotebookEditor == null) {
+    throw new Error("Couldn't find notebook editor for given document");
+  }
 
-    const editorIndex = getCellIndex(editorNotebook, editor.document);
-    const activeEditorIndex = getCellIndex(
-      editorNotebook,
-      activeTextEditor.document
-    );
+  const desiredNotebookDocument: NotebookDocument =
+    desiredNotebookEditor.notebook;
 
-    if (editorIndex === -1 || activeEditorIndex === -1) {
-      throw new Error(
-        "Couldn't find editor corresponding to given cell in the expected notebook"
-      );
-    }
+  await commands.executeCommand(
+    columnFocusCommands[
+      desiredNotebookEditor.viewColumn as keyof typeof columnFocusCommands
+    ]
+  );
 
-    const cellOffset = editorIndex - activeEditorIndex;
+  const activeTextEditor = window.activeTextEditor;
```

Comment:
It seems like we do this in many cases, might be interesting to think if there are ways we can add an AOP type filter for various actions.

Reply:
Sorry not sure I'm following. Fwiw looks like I don't use the active text editor below, so I can remove this null check. It was a holdover from a previous approach. But maybe that's not what you're referring to?